prow/cmd/checkconfig/main.go

```diff
@@ -929,3 +936,49 @@ func validateInRepoConfig(cfg *config.Config, filePath, repoIdentifier string) e
 
 	return nil
 }
+
+func validateTideContextPolicy(cfg *config.Config) error {
+	// We can not know all possible branches without asking GitHub, so instead we verify
+	// all branches that are explicitly configured on any job. This will hopefully catch
+	// most cases.
+	allKnownOrgRepoBranches := map[string]sets.String{}
+	for orgRepo, jobs := range cfg.PresubmitsStatic {
+		if _, ok := allKnownOrgRepoBranches[orgRepo]; !ok {
+			allKnownOrgRepoBranches[orgRepo] = sets.String{}
+		}
+
+		for _, job := range jobs {
+			allKnownOrgRepoBranches[orgRepo].Insert(job.Branches...)
+		}
+	}
+
+	// We have to disableInRepoConfig for this check, else we will
+	// attempt to clone the repo if its enabled
+	originalInRepoConfig := cfg.InRepoConfig
+	cfg.InRepoConfig = config.InRepoConfig{}
+	defer func() { cfg.InRepoConfig = originalInRepoConfig }()
+
+	var errs []error
+	for orgRepo, branches := range allKnownOrgRepoBranches {
+		split := strings.Split(orgRepo, "/")
+		if n := len(split); n != 2 {
+			errs = append(errs, fmt.Errorf("expected to get two results when splitting string %s by '/', got %d", orgRepo, n))
```

Comment:
This assumes that keys to the `presubmits:` section of Prow config are all `org/repo` format which is not true for Gerrit repos. e.g. https://github.com/GoogleCloudPlatform/oss-test-infra/blob/5ddad3c6a30bf826e36c1a9723a71ff7aea80f00/prow/prowjobs/kunit-review.googlesource.com/linux/kunit-linux-config.yaml#L2

This is blocking merge for that entire instance and is a regression so I'm going to revert this PR.

Reply:
https://github.com/kubernetes/test-infra/pull/17467